pandas/core/indexes/datetimes.py

```diff
@@ -2040,7 +2040,11 @@ def date(self):
         Returns numpy array of python datetime.date objects (namely, the date
         part of Timestamps without timezone information).
         """
-        return libts.ints_to_pydatetime(self.normalize().asi8, box="date")
+        if (self.tz is None):
```

Comment:
prob need
``if self.tz is not None and self.tz is not utc``
also write it more like
```
# add a comment here about tz conversions
if self.tz is not None and self.tz is not utc:
    stamps = self.asi8
else:
    stamps = self._local_timestamps()
return libts.init_to_pytdatetime(stamps, box="date")
```